abci/multiplexer.go

```diff
@@ -120,23 +117,24 @@ func NewMultiplexer(
 	return proxy.NewConnSyncLocalClientCreator(wrapper), wrapper.Cleanup, nil
 }
 
-// getAppForHeight gets the appropriate app based on height
-func (m *Multiplexer) getAppForHeight(height int64) (servertypes.ABCI, error) {
+// getApp gets the appropriate app based on height and latest application version.
+func (m *Multiplexer) getApp(height int64) (servertypes.ABCI, error) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
+	// get the appropriate version for this height
+	currentVersion, err := getDesiredVersion(m.lastAppVersion, height, m.versions)
+	if err != nil {
+		// there is no version specified for the given height or application version
+		return m.latestApp, nil
+	}
+
 	// use the latest app if the height is beyond all defined versions
```

Comment:
shouldn't should latest app be updated? it relies only on until height right now.